toy_robot.rb

```diff
@@ -7,34 +7,47 @@ def initialize
     @x = 0
     @y = 0
     @facing = :north
+    @movements = []
   end
 
   def place(x,y,facing)
     raise InvalidPositionError.new("Invalid Position, #{x}, #{y}, #{facing}") if !(x.between?(0,4)) || !(y.between?(0,4))
     @x = x
     @y = y
     @facing = facing
+    @movements.replace(["PLACE #{@x}, #{@y}, #{@facing.to_s.upcase}"])
   end
 
   def move
```

Comment:
This method is getting ugly with all that repetition. Check it in if it's working correctly but you can then refactor it with a separate commit. That is the beauty of the tests, you can refactor with confidence because the tests will tell you if you have made a mistake.